### ISSUE TITLE
[BUG] [4.56.6] Unable to create ec41 storage class: Incompatible version

### ISSUE THREAD

**Issue by onlyjob** [moosefs/moosefs#609]
On MooseFS `4.56.6-1` from Debian (`amd64` architecture), I'm trying to create a new storage class as follows:

```
mfsscadmin  create -M /mnt/mfs  -C C+Q+P,C+P  -K Q+P,P  -A @4+1,M -m STRICT -s 1073741824 -o F  ec41
```

or with even simple command:

```
mfsscadmin  create -M /mnt/mfs   -K Q+P,P  -A @4+1,M   ec41
```

which fails with the following error:

    storage class make ec41: error: Incompatible version

which I would expect _if_ I was trying to use PRO flavours of EC, but according to [mfsscadmin(1)](https://moosefs.com/manpages/mfsscadmin) (and to Release Notes), `@4+1` (EC4+1) should work on CE, is it not?

(If I'm doing something wrong then perhaps `mfsscadmin(1)` man page could be improved by adding example of a command to create such storage class?)

---

P.S. Congratulations upon publishing major release 4, which I have uploaded to Debian "unstable". Great release. Excellent work. Thank you.

**Comment by anon314159** [comment]
Do you have multiple clients connected to your master and are all of your chunkservers/metaloggers/client on the same version?  Generally, speaking you make need to disconnect all clients, upgrade all chunkservers/metaloggers/clients, recycle there associated services and unmount/remount all respective clients.  I ran into the very same issue with client/server version mismatches.

**Comment by onlyjob** [comment]
Everything is on the same version except few legacy v3 clients. Presence of a legacy client should not prevent creation of storage class, IMHO.

**Comment by onlyjob** [comment]
@anon314159, you are correct: after disconnecting last v3 client `mfsscadmin create` successfully created `@4+1` storage class:

    storage class make ec41: ok

I would have never guessed that v3 client was the culprit, as error is the most ambiguous and counter-intuitive.

Thank you.

(I wonder what will happen when some machines that still have v3 client will be powered on and connect to v4 cluster with EC storage class?)

**Comment by onlyjob** [comment]
Thanks, @chogata. IMHO `RESTRICT_INCOMPATIBLE_CLIENT_VERSIONS` should default to `0`, as current default (`1`) is especially brutal to legacy v3 clients: as I have experienced, after defining EC storage class, legacy v3 clients waking up from hibernation/suspend just hang, not being able to register with mfsmaster.

Also, I recommend to improve error thrown by `mfsscadmin`, to highlight "Incompatible version" of what exactly it is complaining about.

**Comment by chogata** [comment]
Yes, the error is already more user-friendly in the description, will appear here with the next batch of commits ;)

Regarding the default - the restriction only applies once you start using the EC format. So during the actual upgrade process nothing will hang. It's a major version upgrade, so keeping any components in a previous version longterm is certainly not recommended, if you really want to do it, you need to inform yourself about the potential consequences. Hence the default you need to consciously change to allow legacy clients to connect.
